news/2019-11-11-WildFly-s2i-openshift-Datasource-configuration.adoc

```diff
@@ -0,0 +1,443 @@
+                <background-validation>true</background-validation>
+                <background-validation-millis>60000</background-validation-millis>
+            </validation>
+        </datasource>
+        <drivers>
+            <driver name="postgresql" module="org.postgresql.jdbc">
+                <xa-datasource-class>org.postgresql.xa.PGXADataSource</xa-datasource-class>
+            </driver>
+        </drivers>
+    </datasources>
+</subsystem>
+----
+
+As you can see in the configuration file, Galleon has prepared the data source subsystem to be configured by the WildFly Datasources Galleon Pack environment variables. You can also verify that a PostgreSQL driver is added as a JBoss module in the server:
+
+[source, bash]
+----
+sh-4.2$ ls /opt/wildfly/modules/org/postgresql/jdbc/main/
+module.xml  postgresql-9.4.1211.jar
+----
+
+This sort of configuration done by using the WildFly Datasource Galleon Pack is simple and easy to use. However, it has some limitations; there are some attributes related to the datasource that cannot be configured, e.g. connection min/max pool size, flush-strategy, background-validation-millis. We cannot configure more than one datasource of the same type. In the following section, we explain how you can achieve this.
+
+Before moving to the next part, let us remove all unused configurations:
+
+[source, bash]
+----
+$ oc delete all -l app=wildfly-app
+pod "wildfly-app-1-84lh6" deleted
+replicationcontroller "wildfly-app-1" deleted
+service "wildfly-app" deleted
+deploymentconfig.apps.openshift.io "wildfly-app" deleted
+buildconfig.build.openshift.io "wildfly-app" deleted
+build.build.openshift.io "wildfly-app-1" deleted
+imagestream.image.openshift.io "wildfly-app" deleted
+route.route.openshift.io "wildfly-app" deleted
+----
+
+== Configuring additional aspects of the datasource subsystem
+
+If you need to configure more than one data source or you need to configure some attributes that are not available by the https://github.com/wildfly-extras/wildfly-datasources-galleon-pack[WildFly Datasources Galleon Pack], there is a generic datasources subsystem configuration by using environment variables. You can check the https://github.com/wildfly/wildfly-s2i/blob/master/doc/datasources.md[Datasources configuration] where these variables are explained. In the next example, we make use of some environments to configure two different datasources specifying different max/min pool sizes.
+
+On detail we need to take into account is we no longer need the PostgreSQL datasource configuration added by the WildFly Datasources Galleon Pack, we are going to configure the data source using a different set of variables, but we still need the PostgreSQL driver added by this Galleon Pack. The solution is easy, just instead of specifying the `postgresql-datasource` layer, we will specify this  the `postgresql-driver` layer which is the layer that brings in only the driver.
```

Comment:
s/On detail/One detail/g
s/ Galleon Pack, we/Galleon Pack, since we/g
s/variables, but/variables. But/g
s/this the/the/g